content/concepts/configuration.md

```diff
@@ -0,0 +1,171 @@
+**webpack.config.ts**
+```typescript
+import * as webpackMerge from 'webpack-merge';
+import { CliConfig } from './config';
+import {
+  getWebpackCommonConfig,
+  getWebpackDevConfigPartial,
+  getWebpackProdConfigPartial,
+  getWebpackMobileConfigPartial,
+  getWebpackMobileProdConfigPartial
+} from './';
+
+export class NgCliWebpackConfig {
+  // TODO: When webpack2 types are finished lets replace all these any types
+  // so this is more maintainable in the future for devs
+  public config: any;
+  private webpackDevConfigPartial: any;
+  private webpackProdConfigPartial: any;
+  private webpackBaseConfig: any;
+  private webpackMobileConfigPartial: any;
+  private webpackMobileProdConfigPartial: any;
+
+  constructor(public ngCliProject: any, public target: string, public environment: string, outputDir?: string) {
+    const config: CliConfig = CliConfig.fromProject();
+    const appConfig = config.config.apps[0];
+
+    appConfig.outDir = outputDir || appConfig.outDir;
+
+    this.webpackBaseConfig = getWebpackCommonConfig(this.ngCliProject.root, environment, appConfig);
+    this.webpackDevConfigPartial = getWebpackDevConfigPartial(this.ngCliProject.root, appConfig);
+    this.webpackProdConfigPartial = getWebpackProdConfigPartial(this.ngCliProject.root, appConfig);
+
+    if (appConfig.mobile){
+      this.webpackMobileConfigPartial = getWebpackMobileConfigPartial(this.ngCliProject.root, appConfig);
+      this.webpackMobileProdConfigPartial = getWebpackMobileProdConfigPartial(this.ngCliProject.root, appConfig);
+      this.webpackBaseConfig = webpackMerge(this.webpackBaseConfig, this.webpackMobileConfigPartial);
+      this.webpackProdConfigPartial = webpackMerge(this.webpackProdConfigPartial, this.webpackMobileProdConfigPartial);
+    }
+
+    this.generateConfig();
+  }
+
+  generateConfig(): void {
+    switch (this.target) {
+      case "development":
+        this.config = webpackMerge(this.webpackBaseConfig, this.webpackDevConfigPartial);
+        break;
+      case "production":
+        this.config = webpackMerge(this.webpackBaseConfig, this.webpackProdConfigPartial);
+        break;
+      default:
+        throw new Error("Invalid build target. Only 'development' and 'production' are available.");
+        break;
+    }
+  }
+}
+```
+
+### Using JSX (React JS Markup) and Babel to create a JSON Configuration that webpack can understand. (Courtesy of [Jason Miller](https://twitter.com/_developit))
+
+```javascript
+import h from 'jsxobj';
+
+/** @jsx h */
+function h(name, attrs, ...children) {
+  let obj = { name, ...attrs };
+  if (typeof name==='function') obj = name(obj);
+  [].concat(...children).forEach( child => {
+    let { name } = child;
+    if (name) {
+      obj[name] = child;
+      delete child.name;
+    }
+    else obj.value = child;
+  });
+  return obj;
+}
+
+// example of an import'd plugin
+const CustomPlugin = config => ({
+  ...config,
+  name: 'custom-plugin'
+});
+
+const CONFIG = (
```

Comment:
Ahh well I suppose I these are examples. I like `config` for _our provided examples_ I didn't want to modify the samples for this page. (Unless you think it will be confusing).